project_generator/tools/uvision.py

```diff
@@ -222,7 +225,7 @@ def _uvproj_set_TargetArmAds(self, uvproj_dic, project_dic):
                     uvproj_dic[self.FLAGS_TO_UVISION[misc_keys]]['Misc'] += ' ' + item
             else:
                 for item in project_dic['misc'][misc_keys]:
-                    uvproj_dic[self.FLAGS_TO_UVISION[misc_keys]]['VariousControls']['MiscControls'] += ' ' + item
+                    uvproj_dic[self.FLAGS_TO_UVISION[misc_keys.encode('utf-8')]]['VariousControls']['MiscControls'] += ' '.encode('utf-8') + item.encode('utf-8')
```

Comment:
``misc_keys.encode('utf-8')`` not needed as a key in the dictionary ;) Otherwise it makes a key b`name`